File: onboarder/src/main/java/com/chyrta/onboarder/GetStartedButton.java
package com.chyrta.onboarder;

import android.support.annotation.ColorRes;
import android.support.annotation.StringRes;

public class GetStartedButton {

    private String title;

    public String getTitle() {
        return title;
    }

    @StringRes
    private int titleResourceId;

    @ColorRes
    private int textColorResourceId;

    public int getTitleResourceId() {
        return titleResourceId;
    }

    public int getTextColorResourceId() {
        return textColorResourceId;
    }

    public static Creator newCreator() {
        return new GetStartedButton().new Creator();
    }

    public class Creator {

        private Creator() {}

        public Creator text(String title) {
            GetStartedButton.this.title = title;
            return this;
        }

        public Creator textResourceId(@StringRes int titleResourceId) {
            GetStartedButton.this.titleResourceId = titleResourceId;
            return this;
        }

        public Creator textColorResourceId(@ColorRes int textColorResourceId) {
            GetStartedButton.this.textColorResourceId = textColorResourceId;
            return this;
        }

        public GetStartedButton create() {
            return GetStartedButton.this;
        }

    }

}

File: onboarder/src/main/java/com/chyrta/onboarder/GetStartedButton.kt
package com.chyrta.onboarder

import android.support.annotation.ColorRes
import android.support.annotation.StringRes

class GetStartedButton {

    var title: String? = null
    var titleResourceId: Int? = null
        private set
        @StringRes get
    var textColorResourceId: Int? = null
        private set
        @ColorRes get

    inner class Creator internal constructor() {

        fun text(title: String): Creator {
            this@GetStartedButton.title = title
            return this
        }

        fun textResourceId(@StringRes titleResourceId: Int): Creator {
            this@GetStartedButton.titleResourceId = titleResourceId
            return this
        }

        fun textColor(@ColorRes textColorResourceId: Int): Creator {
            this@GetStartedButton.textColorResourceId = textColorResourceId
            return this
        }

        fun create(): GetStartedButton {
            return this@GetStartedButton
        }

    }

    companion object {
        fun newCreator(): Creator {
            return GetStartedButton().Creator()
        }
    }

}
